pkg/diagnostics/client.go

```diff
@@ -38,3 +45,126 @@ func (c *Client) Units(node string) (*UnitsHealthResponseJSONStruct, error) {
 		return nil, fmt.Errorf("HTTP %d error", resp.StatusCode)
 	}
 }
+
+// CancelBundle cancel a running job creating a bundle.
+func (c *Client) CancelBundle() (*BundleGenericResponseJSONStruct, error) {
+	resp, err := c.http.Post("/system/health/v1/report/diagnostics/cancel", "application/json", nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 200:
+		var cancelBundle BundleGenericResponseJSONStruct
+		err = json.NewDecoder(resp.Body).Decode(&cancelBundle)
+		return &cancelBundle, err
+	case 503:
+		var cancelBundle BundleGenericResponseJSONStruct
+		err = json.NewDecoder(resp.Body).Decode(&cancelBundle)
+		if err != nil {
+			return nil, err
+		}
+		return nil, errors.New(cancelBundle.Status)
+	default:
+		return nil, fmt.Errorf("HTTP %d error", resp.StatusCode)
+	}
+}
+
+// CreateBundle requests a bundle creation for given nodes on the cluster.
+func (c *Client) CreateBundle(nodes []string) (*BundleCreateResponseJSONStruct, error) {
+	bundleDelete := BundleCreate{Nodes: nodes}
+	var reqBody bytes.Buffer
+	if err := json.NewEncoder(&reqBody).Encode(bundleDelete); err != nil {
+		return nil, err
+	}
+
+	resp, err := c.http.Post("/system/health/v1/report/diagnostics/create", "application/json", &reqBody)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 200:
+		var createdBundle BundleCreateResponseJSONStruct
+		err = json.NewDecoder(resp.Body).Decode(&createdBundle)
+		return &createdBundle, err
+	case 503:
+		return nil, fmt.Errorf("Requested nodes %v not found", nodes)
+	default:
+		return nil, fmt.Errorf("HTTP %d error", resp.StatusCode)
+	}
+}
+
+// GetBundle returns a given diagnostic bundle.
+func (c *Client) GetBundle(bundle string) (*http.Response, error) {
+	if !strings.HasSuffix(bundle, ".zip") {
+		return nil, fmt.Errorf("Format allowed bundle-*.zip")
+	}
+
+	return c.http.Get("/system/health/v1/report/diagnostics/serve/" + bundle)
+}
+
+// DeleteBundle deletes a diagnostics bundle in the cluster.
+func (c *Client) DeleteBundle(bundle string) (*BundleGenericResponseJSONStruct, error) {
```

Comment:
`Delete()`? As diagnostics bundles are the primary entity for this API it seems implicit. For example there is another method called `List()` and not `ListBundles()`.

Reply:
`Create()` and `Cancel()`? for the same reasons